alphadia/workflow/optimization.py

```diff
@@ -260,6 +234,58 @@ def _check_convergence(self):
             < 1.1 * self.history_df[self.feature_name].iloc[-3]
         )
 
+    def _find_index_of_optimum(self):
+        """Finds the index of the row in the history dataframe with the optimal value of the feature used for optimization.
+
+        Returns
+        -------
+        int
+            The index of the row with the optimal value of the feature used for optimization.
+
+        Notes
+        -----
+            This method may be overwritten in child classes if the "optimal" value differs from the value that maximizes the feature used for optimization.
+
+        """
+        return self.history_df[self.feature_name].idxmax()
+
+    def _update_optimization_manager(self):
+        """Updates the optimization manager with the results of the optimization, namely:
+            the classifier version,
+            the optimal parameter,
+            score cutoff,
+            FWHM_RT,
+            and FWHM_mobility
+        at the optimal parameter.
+
+        """
+        index_of_optimum = self._find_index_of_optimum()
+
+        classifier_version_at_optimum = self.history_df["classifier_version"].loc[
+            index_of_optimum
+        ]
+        self.workflow.optimization_manager.fit(
+            {"classifier_version": classifier_version_at_optimum}
+        )
+
+        optimal_parameter = self.history_df["parameter"].loc[index_of_optimum]
+        self.workflow.optimization_manager.fit({self.parameter_name: optimal_parameter})
```

Comment:
(nit^3) 
turning the order of the first two (also in docstring) would make it even more readable :-)
```
        optimal_parameter = self.history_df["parameter"].loc[index_of_optimum]
        self.workflow.optimization_manager.fit({self.parameter_name: optimal_parameter})

        classifier_version_at_optimum = self.history_df["classifier_version"].loc[
            index_of_optimum
        ]
        self.workflow.optimization_manager.fit(
            {"classifier_version": classifier_version_at_optimum}
        )
```